packages/common/utils/src/array/NonEmptyArray.en.md

```diff
@@ -1,3 +1,25 @@
 # NonEmptyArray
 
-A type which indicates that an array is not empty. We can assert an array not to be empty by `isNonEmptyArray` function.
+A type which indicates that an array is not empty.
+
+```typescript
+type NonEmptyArray<T> = [T, ...T[]];
+```
+
+- You can use this to create a function like [isNonEmptyArray](https://slash.page/ko/libraries/common/utils/src/array/isnonemptyarray.i18n/) that checks if an array has at least one element.
+
+# Example
+
+```typescript
+// There is no problem at the time of declaration, but runtime errors may occur depending on the internal logic.
+const getSum = (array: number[]) => array.reduce((x, y) => x + y);
+
+// [ERR]: Reduce of empty array with no initial value
+getSum([]);
+
+// Runtime errors can be prevented through type errors at the time of declaration.
+const getSum = (array: NonEmptyArray<number>) => array.reduce((x, y) => x + y);
+
+// Argument of type '[]' is not assignable to parameter of type 'NonEmptyArray<number>'. Source has 0 element(s) but target requires
+getSum([]);
+```
```

Comment:
Since `sum` in our libraries are defined as `arr.reduce((x, y) => x + y, 0)`, I thought this example might be a little confusing. 

Why not use a more easy example like `getFirstElement()`? 

```ts
function getFirstElement<T>(arr: T[]): T {
  return arr[0];
}

const items: number[] = [];

// Runtime error
getFirstElement([]).toString()
```

Reply:
@raon0211 thank you I thought it was too difficult!
I modified the code as below and added it to the example.
```typescript
function getFirstElement<T>(arr: T[]): T {
  return arr[0];
}

const items: number[] = [];

// Runtime error
getFirstElement([]).toString();

// Type error
const fixItems: NonEmptyArray<number> = [];
```
Thank you for checking.